### ISSUE TITLE
How to specify an unsubscribe group to associate the email?

### ISSUE THREAD

**Issue by IraklisElef** [Jakebreen/android-sendgrid#20]
Hello,

As the question suggests, I am trying to find a way to specify an unsubscribe group to associate with the email. Is there any way to do that? Thank you in advance!

Sincerely,
Iraklis Eleftheriadis

**Comment by jakebreen** [comment]
Hello, this is not currently a feature in the library...

I see in the API an `asm` object can be sent as part of the mail's request body, is this what you had in mind?

https://docs.sendgrid.com/api-reference/mail-send/mail-send

```
asm
object

An object allowing you to specify how to handle unsubscribes.
group_id
integer

The unsubscribe group to associate with this email.
required
groups_to_display
array[integer]

An array containing the unsubscribe groups that you would like to be displayed on the unsubscribe preferences page.
maxItems: 25
```

**Comment by IraklisElef** [comment]
Yes that's exactly what I need. In order for the "Unsubscribe" link to work at the bottom of the email, I need to specify the asm parameter with the group_id of the particular unsubscribe group. If I do not include that parameter, the link is simply not clickable. (Which is vital in order to comply with spam laws)

**Comment by jakebreen** [comment]
You are welcome to implement it and create a PR for me to review otherwise I'll do it at some point this week if I have time.

**Comment by IraklisElef** [comment]
Any updates maybe?

**Comment by jakebreen** [comment]
I've started to take a look at it.

**Comment by IraklisElef** [comment]
Thank you mate!

**Comment by jakebreen** [comment]
I've created the feature and it will be available in version 1.4.0

Added two new methods for setting unsubscribe group(s):

`setUnsubscribeGroupId(@NonNull int groupId)`
`setUnsubscribeGroupIds(@NonNull int groupId, List<Integer> groupIds)`

Only one of these methods needs to be used depending on whether or not you want to provide a list of groups long with the unsubscribe group id.

Let me know how it goes.

**Comment by IraklisElef** [comment]
I've just tested your new version and it works exactly like I needed it to, thank you again!
